solr/core/src/java/org/apache/solr/schema/BinaryField.java

```diff
@@ -112,7 +121,36 @@ public IndexableField createField(SchemaField field, Object val) {
       len = buf.length;
     }
 
-    return new org.apache.lucene.document.StoredField(field.getName(), buf, offset, len);
+    return new BytesRef(buf, offset, len);
+  }
+
+  @Override
+  public List<IndexableField> createFields(SchemaField field, Object val) {
```

Comment:
This is pretty much borrowed from `StrField` type that supports creating `stored` and/or `docValued` Lucene fields for a single Solr field.